staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -936,11 +936,10 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, storagePolicyName string, nodeN
 			return "", err
 		}
 
-		// try and get canonical path for disk and if we can't throw error
-		vmDiskPath, err = getcanonicalVolumePath(ctx, vm.Datacenter, vmDiskPath)
-		if err != nil {
-			klog.Errorf("failed to get canonical path for %s on node %s: %v", vmDiskPath, convertToString(nodeName), err)
-			return "", err
+		// try and get canonical path for disk and if we can't use provided vmDiskPath
+		canonicalPath, pathFetchErr := getcanonicalVolumePath(ctx, vm.Datacenter, vmDiskPath)
+		if canonicalPath != "" && pathFetchErr == nil {
+			vmDiskPath = canonicalPath
```

Comment:
Does it matter what the error was? At least one of the possible errors seems to be a failure to parse the vmDiskPath. Would you still want to throw an error in that case?

Reply:
Validating anything here could be a regression and hence I am leaving the error alone and using original disk path present in PV if we can't get canonical path. It is upto backend then to decide if it can attach the disk.  This still gets us the fix we wanted in original PR.